File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxSum = nums[0]
        sum = 0
        
        for n in nums:
            if sum < 0:
                sum = 0
            sum +=n
            maxSum = max(sum, maxSum)

        return maxSum

File: 53-maximum-subarray/53-maximum-subarray.rs
impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut curr_sum = 0;
        let mut max_sum = i32::MIN;
        
        for n in nums{
            curr_sum += n;
            
            if curr_sum > max_sum {
                max_sum = curr_sum;
            }
            
            if curr_sum < 0{
                curr_sum = 0;
            }
        }
        
        max_sum
    }
}